p2p/host/basic/natmgr.go

```diff
@@ -25,8 +25,8 @@ type NATManager interface {
 }
 
 // NewNATManager creates a NAT manager.
-func NewNATManager(net network.Network) NATManager {
-	return newNATManager(net)
+func NewNATManager(net network.Network, userAgent string) NATManager {
```

Comment:
Nicer: use the option pattern.

Reply:
@marten-seemann What does that mean? In what place? Where is it better to take the variable from?

Reply:
@marten-seemann Ok. Maybe this is the right way?

Reply:
Marten likely meant defining a new options type for the `NATManager` instead of extending Swarm's options set and Network interface with UserAgant getter.

Reply:
@Wondertan Why not?

Reply:
Managing UserAgent is logically not Network's responsibility.

Reply:
@Wondertan Whose responsibility is that?

Reply:
IIRC, it's just a param that's passed from BasicHost down to Identify, so you can intercept it and pass it to discoverNAT as an option.

Reply:
@marten-seemann Where is the answer from Marten?

Reply:
@Wondertan 

> // Network is the interface used to connect to the outside world.
// It dials and listens for connections.

I use Network to store my UserAgent so that Network can tell the outside world my UserAgent. Isn't that logical?

Reply:
I was talking about the option pattern, like here: https://github.com/libp2p/go-libp2p/blob/master/p2p/protocol/circuitv2/relay/options.go

Reply:
@marten-seemann And you think [putting the UserAgent in the Swarm](https://github.com/libp2p/go-libp2p/pull/2310/files#diff-562d33c4dff7fb735c38df99f47ca60a7ec4043be9199d5ca081745f6239a09cR36) - Network is not right?

Reply:
@marten-seemann I don't know. I just read the comments about what the swarm was doing and looked at which direction the data was going. If the Network is establishing connections, it needs a UserAgent.

Okay, if this idea of mine is wrong, I'll fix the code. I'll try to make an options.

Reply:
@marten-seemann The new version with options is ready. But it seems to me it is too much for one parameter.